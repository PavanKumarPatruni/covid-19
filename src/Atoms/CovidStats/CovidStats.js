/* eslint-disable no-unused-vars */
import React from 'react';
import PropTypes from 'prop-types';

import './CovidStats.scss';

const propTypes = {
  stats: PropTypes.object,
};

const CovidStats = ({ stats }) => {
  const getData = (item, key) => {
    return item && item[key];
  };

  const getDeltaComponent = value => {
    return value > 0 ? (
      <p className="delta-count mt2">{`[+${value}]`}</p>
    ) : (
      <p className="delta-count mt2"></p>
    );
  };

  const getStat = item => {
    return (
      <div
        className={`flex flex-column flex-justify-center flex-center covid-stats-item p1 sm-p0 sm-mx1 md-mx2 mx4 ${item.className}`}
      >
        <h3 className="title uppercase mt2">{item.title}</h3>
        {getDeltaComponent(item.delta)}
        <p className="total-count mt1">{item.total}</p>
      </div>
    );
  };

  return (
    <div className="flex covid-stats p1 comp-mb">
      {getStat({
        title: 'Confirmed',
        total: getData(stats, 'confirmed'),
        delta: getData(stats, 'deltaconfirmed'),
        className: 'red-color',
      })}
      {getStat({
        title: 'Active',
        total: getData(stats, 'active'),
        delta: 0,
        className: 'blue-color',
      })}
      {getStat({
        title: 'Recovered',
        total: getData(stats, 'recovered'),
        delta: getData(stats, 'deltarecovered'),
        className: 'green-color',
      })}
      {getStat({
        title: 'Deceased',
        total: getData(stats, 'deaths'),
        delta: getData(stats, 'deltadeaths'),
        className: 'voilet-color',
      })}
    </div>
  );
};

CovidStats.defaultProps = {
  stats: {},
};

CovidStats.propTypes = propTypes;

export default CovidStats;
