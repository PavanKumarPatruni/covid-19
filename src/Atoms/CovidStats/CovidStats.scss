@import '../../styles/index.scss';

.delta-count {
  height: 20px;
}

.title {
  text-transform: uppercase;
}

.red-color {
  .total-count {
    @include font-size-32-bold($red-color);

    @include mobile {
      @include font-size-20-bold($red-color);
    }
  }

  .delta-count {
    @include font-size-20-bold($red-color);

    @include mobile {
      @include font-size-16-bold($red-color);
    }
  }

  .title {
    @include font-size-tag($red-color);

    @include mobile {
      @include font-size-tag-mobile($red-color);
    }
  }
}

.green-color {
  .total-count {
    @include font-size-32-bold($green-color);

    @include mobile {
      @include font-size-20-bold($green-color);
    }
  }

  .delta-count {
    @include font-size-20-bold($green-color);

    @include mobile {
      @include font-size-16-bold($green-color);
    }
  }

  .title {
    @include font-size-tag($green-color);

    @include mobile {
      @include font-size-tag-mobile($green-color);
    }
  }
}

.blue-color {
  .total-count {
    @include font-size-32-bold($blue-color);

    @include mobile {
      @include font-size-20-bold($blue-color);
    }
  }

  .delta-count {
    @include font-size-20-bold($blue-color);

    @include mobile {
      @include font-size-16-bold($blue-color);
    }
  }

  .title {
    @include font-size-tag($blue-color);

    @include mobile {
      @include font-size-tag-mobile($blue-color);
    }
  }
}

.grey-color {
  .total-count {
    @include font-size-32-bold($tertiary-color);

    @include mobile {
      @include font-size-20-bold($tertiary-color);
    }
  }

  .delta-count {
    @include font-size-20-bold($tertiary-color);

    @include mobile {
      @include font-size-16-bold($tertiary-color);
    }
  }

  .title {
    @include font-size-tag($tertiary-color);

    @include mobile {
      @include font-size-tag-mobile($tertiary-color);
    }
  }
}
