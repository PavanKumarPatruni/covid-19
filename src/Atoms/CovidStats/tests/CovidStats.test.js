import React from 'react';
import { shallow, mount } from 'enzyme';

import CovidStats from '../CovidStats';

describe('CovidStats component', () => {
  const props = {
    stats: {
      active: '44351',
      confirmed: '67700',
      deaths: '2215',
      deltaconfirmed: '524',
      deltadeaths: '2',
      deltarecovered: '160',
      lastupdatedtime: '11/05/2020 15:41:22',
      recovered: '21130',
      state: 'Total',
      statecode: 'TT',
      statenotes: '',
    },
  };

  test('match snapshot correctly', () => {
    const wrapper = shallow(<CovidStats {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('match snapshot correctly', () => {
    const wrapper = mount(<CovidStats {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
