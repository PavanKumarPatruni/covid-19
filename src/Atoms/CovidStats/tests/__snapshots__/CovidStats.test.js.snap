// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CovidStats component match snapshot correctly 1`] = `
<div
  className="flex covid-stats p1 comp-mb"
>
  <div
    className="flex flex-column flex-justify-center flex-center covid-stats-item p1 sm-p0 sm-mx1 md-mx2 mx4 red-color"
  >
    <h3
      className="title mt2"
    >
      Confirmed
    </h3>
    <p
      className="delta-count mt2"
    >
      [+524]
    </p>
    <p
      className="total-count mt1"
    >
      67700
    </p>
  </div>
  <div
    className="flex flex-column flex-justify-center flex-center covid-stats-item p1 sm-p0 sm-mx1 md-mx2 mx4 blue-color"
  >
    <h3
      className="title mt2"
    >
      Active
    </h3>
    <p
      className="delta-count mt2"
    />
    <p
      className="total-count mt1"
    >
      44351
    </p>
  </div>
  <div
    className="flex flex-column flex-justify-center flex-center covid-stats-item p1 sm-p0 sm-mx1 md-mx2 mx4 green-color"
  >
    <h3
      className="title mt2"
    >
      Recovered
    </h3>
    <p
      className="delta-count mt2"
    >
      [+160]
    </p>
    <p
      className="total-count mt1"
    >
      21130
    </p>
  </div>
  <div
    className="flex flex-column flex-justify-center flex-center covid-stats-item p1 sm-p0 sm-mx1 md-mx2 mx4 grey-color"
  >
    <h3
      className="title mt2"
    >
      Deceased
    </h3>
    <p
      className="delta-count mt2"
    >
      [+2]
    </p>
    <p
      className="total-count mt1"
    >
      2215
    </p>
  </div>
</div>
`;

exports[`CovidStats component match snapshot correctly 2`] = `
<CovidStats
  stats={
    Object {
      "active": "44351",
      "confirmed": "67700",
      "deaths": "2215",
      "deltaconfirmed": "524",
      "deltadeaths": "2",
      "deltarecovered": "160",
      "lastupdatedtime": "11/05/2020 15:41:22",
      "recovered": "21130",
      "state": "Total",
      "statecode": "TT",
      "statenotes": "",
    }
  }
>
  <div
    className="flex covid-stats p1 comp-mb"
  >
    <div
      className="flex flex-column flex-justify-center flex-center covid-stats-item p1 sm-p0 sm-mx1 md-mx2 mx4 red-color"
    >
      <h3
        className="title mt2"
      >
        Confirmed
      </h3>
      <p
        className="delta-count mt2"
      >
        [+524]
      </p>
      <p
        className="total-count mt1"
      >
        67700
      </p>
    </div>
    <div
      className="flex flex-column flex-justify-center flex-center covid-stats-item p1 sm-p0 sm-mx1 md-mx2 mx4 blue-color"
    >
      <h3
        className="title mt2"
      >
        Active
      </h3>
      <p
        className="delta-count mt2"
      />
      <p
        className="total-count mt1"
      >
        44351
      </p>
    </div>
    <div
      className="flex flex-column flex-justify-center flex-center covid-stats-item p1 sm-p0 sm-mx1 md-mx2 mx4 green-color"
    >
      <h3
        className="title mt2"
      >
        Recovered
      </h3>
      <p
        className="delta-count mt2"
      >
        [+160]
      </p>
      <p
        className="total-count mt1"
      >
        21130
      </p>
    </div>
    <div
      className="flex flex-column flex-justify-center flex-center covid-stats-item p1 sm-p0 sm-mx1 md-mx2 mx4 grey-color"
    >
      <h3
        className="title mt2"
      >
        Deceased
      </h3>
      <p
        className="delta-count mt2"
      >
        [+2]
      </p>
      <p
        className="total-count mt1"
      >
        2215
      </p>
    </div>
  </div>
</CovidStats>
`;
